<h2>Building a classifier</h2>

In [55]:
from sklearn.model_selection import train_test_split
from sklearn.metrics import confusion_matrix
from sklearn.metrics import classification_report
X_train, X_test, y_train, y_test = train_test_split(X_new, y_new, test_size=0.3, random_state=0)

In [57]:
"""Using Decision Tree Classifer"""
from sklearn.tree import DecisionTreeClassifier
dt = DecisionTreeClassifier()
dt.fit(X_train, y_train)
y_pred = dt.predict(X_test)
print('Accuracy of Decision Tree classifier on test set: {:.2f}'.format(dt.score(X_test, y_test)))
from sklearn.metrics import confusion_matrix
confusion_matrix = confusion_matrix(y_test, y_pred)
print(confusion_matrix)
print(classification_report(y_test, y_pred))

Accuracy of Decision Tree classifier on test set: 0.93
[[637  56]
 [ 41 624]]
              precision    recall  f1-score   support

           0       0.94      0.92      0.93       693
           1       0.92      0.94      0.93       665

    accuracy                           0.93      1358
   macro avg       0.93      0.93      0.93      1358
weighted avg       0.93      0.93      0.93      1358



In [58]:
"""Using KNN Classifer"""
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors=3)
knn.fit(X_train, y_train)
y_pred = knn.predict(X_test)
print('Accuracy of Knn classifier on test set: {:.2f}'.format(knn.score(X_test, y_test)))
from sklearn.metrics import confusion_matrix
confusion_matrix = confusion_matrix(y_test, y_pred)
print(confusion_matrix)
from sklearn.metrics import classification_report
print(classification_report(y_test, y_pred))

Accuracy of Knn classifier on test set: 0.85
[[511 182]
 [ 26 639]]
              precision    recall  f1-score   support

           0       0.95      0.74      0.83       693
           1       0.78      0.96      0.86       665

    accuracy                           0.85      1358
   macro avg       0.86      0.85      0.85      1358
weighted avg       0.87      0.85      0.85      1358



In [59]:
"""Using Support Vector Classifer"""
from sklearn.svm import SVC
svm_classifier = SVC(gamma='auto')
svm_classifier.fit(X_train, y_train)
y_pred = svm_classifier.predict(X_test)
print('Accuracy of Support Vector classifier on test set: {:.2f}'.format(svm_classifier.score(X_test, y_test)))
from sklearn.metrics import confusion_matrix
confusion_matrix = confusion_matrix(y_test, y_pred)
print(confusion_matrix)
from sklearn.metrics import classification_report
print(classification_report(y_test, y_pred))

Accuracy of Support Vector classifier on test set: 0.92
[[654  39]
 [ 66 599]]
              precision    recall  f1-score   support

           0       0.91      0.94      0.93       693
           1       0.94      0.90      0.92       665

    accuracy                           0.92      1358
   macro avg       0.92      0.92      0.92      1358
weighted avg       0.92      0.92      0.92      1358



Support Vector and Decision Tree classifiers are classifying more accurately compared to other classifiers with F1 scores = 0.93